aws/resource_aws_cognito_user_pool.go

```diff
@@ -0,0 +1,902 @@
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"email_message": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validateCognitoUserPoolInviteTemplateEmailMessage,
+									},
+									"email_subject": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validateCognitoUserPoolTemplateEmailSubject,
+									},
+									"sms_message": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validateCognitoUserPoolInviteTemplateSmsMessage,
+									},
+								},
+							},
+						},
+						"unused_account_validity_days": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      7,
+							ValidateFunc: validateIntegerInRange(0, 90),
+						},
+					},
+				},
+			},
+
+			"alias_attributes": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validateCognitoUserPoolAliasAttribute,
+				},
+				ConflictsWith: []string{"username_attributes"},
+			},
+
+			"auto_verified_attributes": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validateCognitoUserPoolAutoVerifiedAttribute,
+				},
+			},
+
+			"creation_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"device_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MinItems: 0,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"challenge_required_on_new_device": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"device_only_remembered_on_user_prompt": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"email_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MinItems: 0,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"reply_to_email_address": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validateCognitoUserPoolReplyEmailAddress,
+						},
+						"source_arn": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validateArn,
+						},
+					},
+				},
+			},
+
+			"email_verification_subject": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validateCognitoUserPoolEmailVerificationSubject,
+			},
+
+			"email_verification_message": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validateCognitoUserPoolEmailVerificationMessage,
+			},
+
+			"lambda_config": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
```

Comment:
Does the API also set bunch of default Lambda functions? 😮

Reply:
Nupe, you're right :D fixing it!

Reply:
By bunch: do you mean declaring several lambda_config sections? There is the `MaxItems: 1` defined

Reply:
right, I meant 1 Lambda function ARN per purpose.

Reply:
Oh, then... yes! even though the documentation for attributes is not always crystal-clear, those are alllambda functions... (checked against the UI, Cloudformation parameters & the rest API)